types/soap/index.d.ts

```diff
@@ -1,87 +1,188 @@
+    // wsdl options that only work for client
+    forceSoap12Headers?: boolean;
+    customDeserializer?: any;
+    [key: string]: any;
+}
+
+export interface IServerOptions extends IWSDLBaseOptions {
+    path: string;
+    services: IServices;
+    xml?: string;
+    uri?: string;
+    suppressStack?: boolean;
+    [key: string]: any;
 }
 
-export class ClientSSLSecurity implements Security {
-    constructor(key: Buffer | string, cert: Buffer | string, ca: Buffer | string, defaults?: any);
-    constructor(key: Buffer | string, cert: Buffer | string, defaults?: any);
+export class WSDL {
+    constructor(definition: any, uri: string, options?: IOptions);
+    ignoredNamespaces: string[];
+    ignoreBaseNameSpaces: boolean;
+    valueKey: string;
+    xmlKey: string;
+    xmlnsInEnvelope: string;
+    onReady(callback: (err:Error) => void): void;
+    processIncludes(callback: (err:Error) => void): void;
+    describeServices(): { [k: string]: any };
+    toXML(): string;
+    xmlToObject(xml: any, callback?: (err:Error, result:any) => void): any;
+    findSchemaObject(nsURI: string, qname: string): any;
+    objectToDocumentXML(name: string, params: any, nsPrefix?: string, nsURI?: string, type?: string): any;
+    objectToRpcXML(name: string, params: any, nsPrefix?: string, nsURI?: string, isParts?: any): string;
+    isIgnoredNameSpace(ns: string): boolean;
+    filterOutIgnoredNameSpace(ns: string): string;
+    objectToXML(obj: any, name: string, nsPrefix?: any, nsURI?: string, isFirst?: boolean, xmlnsAttr?: any, schemaObject?: any, nsContext?: any): string;
+    processAttributes(child: any, nsContext: any): string;
+    findSchemaType(name: any, nsURI: any): any;
+    findChildSchemaObject(parameterTypeObj: any, childName: any, backtrace?: any): any;
 }
 
-export interface Client extends events.EventEmitter {
+export class Client extends EventEmitter {
+    constructor(wsdl: WSDL, endpoint?: string, options?: IOptions);
     addBodyAttribute(bodyAttribute: any, name?: string, namespace?: string, xmlns?: string): void;
     addHttpHeader(name: string, value: any): void;
-    addSoapHeader(soapHeader: any, name?: string, namespace?: string, xmlns?: string): number;
+    addSoapHeader(soapHeader: any, name?: string, namespace?: any, xmlns?: string): number;
     changeSoapHeader(index: number, soapHeader: any, name?: string, namespace?: string, xmlns?: string): void;
     clearBodyAttributes(): void;
     clearHttpHeaders(): void;
     clearSoapHeaders(): void;
     describe(): any;
     getBodyAttributes(): any[];
-    getHttpHeaders(): any[];
-    getSoapHeaders(): any[];
+    getHttpHeaders(): { [k:string]: string };
+    getSoapHeaders(): string[];
     setEndpoint(endpoint: string): void;
     setSOAPAction(action: string): void;
-    setSecurity(s: Security): void;
-    [method: string]: SoapMethod | Function;
+    setSecurity(security: ISecurity): void;
+    [key: string]: any; // [method: string]: ISOAPMethod | Function;
```

Comment:
> it should be [key: string]: any; otherwise TS will complain on any member you add to the extended class if it doesn't have a type of SoapMethod | Function.

Can you give example code of extended class? I tried to write a sub class without changing it to `any` and there was no error reported by TSC.